server/grpc_service.go

```diff
@@ -1796,3 +1801,38 @@ func (s *GrpcServer) sendAllGlobalConfig(ctx context.Context, server pdpb.PD_Wat
 func (s *GrpcServer) ReportBuckets(pdpb.PD_ReportBucketsServer) error {
 	panic("not implemented")
 }
+
+// Evict the leaders when the store is damaged. Damaged regions are emergency errors
+// and requires user to manually remove the `evict-leader-scheduler` with pd-ctl
+func (s *GrpcServer) handleDamagedStore(stats *pdpb.StoreStats) error {
+	// ToDO: regions have no special process for the time being
+	// and need to be removed in the future
+	damagedRegions := stats.GetDamagedRegionsId()
+	if len(damagedRegions) == 0 {
+		return nil
+	}
+
+	log.Error("store damaged and leaders will be evicted, you might fix the store and remove evict-leader-scheduler manually",
+		zap.Uint64("store-id", stats.GetStoreId()),
+		zap.Uint64s("region-ids", damagedRegions))
+
+	h := s.GetHandler()
+	if exist, err := h.IsSchedulerExisted(schedulers.EvictLeaderName); !exist {
+		if err != nil {
+			return err
+		}
+		err = h.AddEvictLeaderScheduler(stats.GetStoreId())
+		if err != nil {
+			return err
+		}
+	} else {
+		if err := h.RedirectSchedulerUpdate(schedulers.EvictLeaderName, float64(stats.GetStoreId())); err != nil {
+			return err
+		}
+		log.Info("update scheduler",
```

Comment:
What about adding some info to help distinguish this part from the same API logs?